src/storage/core.rs

```diff
@@ -433,21 +436,24 @@ impl<K> Storage<K> {
     }
 
     /// `check_bloom` is used to check whether a key is in storage.
+    /// If bloom filter opt out, returns `None`.
     /// Uses bloom filter under the hood, so false positive results are possible,
     /// but false negatives are not.
     /// In other words, `check_bloom` returns either "possibly in storage" or "definitely not".
-    pub async fn check_bloom(&self, key: impl Key) -> bool {
+    pub async fn check_bloom(&self, key: impl Key) -> Option<bool> {
         trace!("[{:?}] check in blobs bloom filter", &key.to_vec());
         let inner = self.inner.safe.lock().await;
         let in_active = inner
             .active_blob
             .as_ref()
-            .map_or(false, |active_blob| active_blob.check_bloom(key.as_ref()));
+            .map_or(Some(false), |active_blob| {
```

Comment:
is there on pearl level options to skip bloom index lookup?

Reply:
Yes, if filter is opt out, `check_bloom` will return `None`.